### ISSUE TITLE
yocto 1.8 and arbor43c

### ISSUE THREAD

**Issue by Flow86** [gumstix/yocto-manifest#63]
Hi,

I tried to use "fido" with an overo storm and an arbor43c board (with newhaven touch screen).

I have some problems:
- touchscreen input does not work at all
  - trying to load "edt_ft5x06" (modprobe) crashes system
- ethernet in u-boot does not work ("Net:   No ethernet found.") 
  - bootp says "No ethernet found."
- ethernet inside yocto only works if cable is connected (otherwise device is simply not existent)

what should I do now?

**Comment by ashcharles** [comment]
The touch screen should load without any need to modprobe the device as long as the correct board is getting booted.  In the boot log ('dmesg'), do you see anything about 'arbor43'?  If not, please try these commands at the u-boot prompt:
# i2c dev 2
# i2c mw 0x51 0 0x00 40
# i2c mw 0x51 1 0x02
# i2c mw 0x51 3 0x0D
# reset

**Comment by Flow86** [comment]
hi,

```
[    0.000000] Machine model: OMAP36xx/AM37xx/DM37xx Gumstix Overo on Arbor43C
```
is the only message I get about arbor in dmesg

the i2c commands havent changed anything

**Comment by Flow86** [comment]
well, Ive downloaded the precompiled images from gumstix.com (which is 1.7.1, not 1.8) - there the touchscreen works (after loading ft5x0x_ts kernel module)

u-boot does find the ethernet, and linux does find the ethernet card if cable is not plugged in.

so something terribly went wrong with 1.8 :-1:

**Comment by Flow86** [comment]
okay, 

1.8
- touchscreen simply does not work
- u-boot 2015.07: ethernet does not work

1.7.1 (precompiled image)
- touchscreen works
- u-boot  2014.10: ethernet works

1.7.1 (custom image)
- touchsreen works
- u-boot 2015.04: ethernet does not work

so: 
- betwen 2014.10 and 2015.04 in u-boot ethernet support seems to be corrupted 
- between 1.7.1 and 1.8 touchscreen driver was corrupted/removed/changed

**Comment by ashcharles** [comment]
Thanks for the clear breakdown---this was helpful to track down the issues.  The ethernet and touchscreen works okay on my Arbo43C using an image compiled from the latest dev (i.e. 1.8) branch.  For the touchscreen, we need:
 * https://github.com/gumstix/meta-gumstix/commit/bb0f3136ffe7e1677333f4bb2248c7b4323e5205
 * and https://github.com/gumstix/linux/commit/922332cda57cb522e2f0990defc35ad530888b7e makes things useful

For ethernet support, the SMSC chip wasn't getting configured for the Arbor43C as I neglected to carry forward support when ethernet initialization changed upstream.  Fixed up here:
https://github.com/gumstix/u-boot/commit/a7f1324e1a5908e2ef5a80c2ead0755567887527
